mlrun/datastore/base.py

```diff
@@ -156,6 +161,24 @@ def to_dict(self):
         }
 
 
+def _drop_reserved_columns(df):
+    cols_to_drop = []
+    reserved_column_names = {
+        "key",
+        "date",
+        "year",
+        "month",
+        "day",
+        "hour",
+        "minute",
+        "second",
+    }
+    for col in df.columns:
+        if col in reserved_column_names or col.startswith("hash_"):
+            cols_to_drop.append(col)
+    df.drop(labels=cols_to_drop, axis=1, inplace=True, errors="ignore")
```

Comment:
need to distinguish between columns we added for partitioning and user data
you can find out the partition columns like we do here - 
https://github.com/mlrun/storey/blob/fb391a23258793dae20724e5b728b381148f5d1e/storey/sources.py#L709

Reply:
Partition columns can be user columns as well. They're not all necessarily artificial.

Reply:
right, that's what i said. the user can have his own column named year and we can't drop it

Reply:
Using a dedicated prefix now.